src/services/supabase.ts

```diff
@@ -107,3 +111,88 @@ export const getUserDetails = (user: User | null) => {
         avatar,
     };
 };
+
+export interface CallSupabaseResponse {
+    error?: PostgrestError;
+    data: any;
+}
+
+const handleSuccess = (response: any) => {
```

Comment:
Is there any reason to not use a single return statement here?

`return response.error ? { data: null, error: response.error } : { data: response.data };`